packages/core/tracing/src/components/waterfall/WaterfallView.vue

```diff
@@ -0,0 +1,331 @@
+//   (e) => {
+//     if (interaction.value !== 'zoom') {
+//       return
+//     }
+
+//     e.event.preventDefault()
+
+//     const sbmRect = getSBMRect()!
+//     if (e.event.x < sbmRect.x) {
+//       return
+//     }
+
+//     if (Math.abs(e.delta[0]) > Math.abs(e.delta[1])) {
+//       const viewportShift = e.delta[0] / config.zoom / sbmRect.width
+//       config.viewport.left += viewportShift
+//       config.viewport.right -= viewportShift
+//       if (config.viewport.left < 0) {
+//         config.viewport.right += config.viewport.left
+//         config.viewport.left = 0
+//       } else if (config.viewport.right < 0) {
+//         config.viewport.left += config.viewport.right
+//         config.viewport.right = 0
+//       }
+//     } else {
+//       const nextZoom = Math.max(1, config.zoom - (e.delta[1] / sbmRect.width) * 4)
+//       const viewportWidth = 1 - config.viewport.left - config.viewport.right
+//       const nextViewportWidth = 1 / nextZoom
+//       const viewportWidthDelta = nextViewportWidth - viewportWidth
+
+//       const zoomOrigin = (e.event.x - sbmRect.x) / sbmRect.width
+
+//       config.viewport.left = config.viewport.left - viewportWidthDelta * zoomOrigin
+//       config.viewport.right = config.viewport.right - viewportWidthDelta * (1 - zoomOrigin)
+
+//       if (config.viewport.left < 0 && config.viewport.right < 0) {
+//         config.viewport.left = 0
+//         config.viewport.right = 0
+//       } else if (config.viewport.left < 0) {
+//         config.viewport.right += config.viewport.left
+//         config.viewport.left = 0
+//         if (config.viewport.right < 0) {
+//           config.viewport.right = 0
+//         }
+//       } else if (config.viewport.right < 0) {
+//         config.viewport.left += config.viewport.right
+//         config.viewport.right = 0
+//         if (config.viewport.left < 0) {
+//           config.viewport.left = 0
+//         }
+//       }
+
+//       config.zoom = 1 / (1 - config.viewport.left - config.viewport.right)
+//     }
+//   },
+//   {
+//     domTarget: rowsAreaRef,
+//     eventOptions: {
+//       passive: false,
+//     },
+//   },
+// )
+
+// RESERVED: Only used when zooming is enabled
+// watchEffect(() => {
+//   const sbmRect = getSBMRect()
+//   if (sbmRect) {
+//     const minViewportShift = -(sbmRect.width * config.zoom) + sbmRect.width
+
+//     if (config.viewportShift > 0) {
+//       config.viewportShift = 0
+//     } else if (config.viewportShift < minViewportShift) {
+//       config.viewportShift = minViewportShift
+//     }
+//   }
+// })
+</script>
+
+<style lang="scss" scoped>
+.waterfall {
+  box-sizing: border-box;
+  height: 100%;
+  overflow-y: v-bind("interaction === 'zoom' ? 'hidden' : 'scroll'");
+
+  :deep(.waterfall-row) {
+    box-sizing: border-box;
+    column-gap: $kui-space-40;
+    display: grid;
+    grid-template-columns: 400px auto;
+    padding: 0 $kui-space-40;
+    width: 100%;
+
+    &> :nth-child(2) {
```

Comment:
```suggestion
    & > :nth-child(2) {
```

Reply:
It looks like "Vue - Official" prefers this but ESLint or Stylelint did not give any opinion on this–I have updated this selector.